LXC doesn't correctly handle sub-command aliases

# Required information

Ubuntu 16.04, LXC 2.0.2

# Issue description

LXC will incorrectly match a sub-command alias due to effectively checking only the first arg of the alias, e.g. given the alias `lxc image cp`, any `lxc image` command will match the alias.

# Steps to reproduce

```
printf "aliases:\n  image cp: image copy\n" >> ~/.config/lxc/config.yml
lxc image list
...
error: wrong number of subcommand arguments
```

This can also manifest as a panic if the original command has fewer arguments than the incorrectly matched alias.

```
10:18:13 ✔ ~ $ lxc image
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
panic(0x8c9320, 0xc82000e100)
	/usr/lib/go-1.6/src/runtime/panic.go:464 +0x3e6
main.expandAlias(0xc8200ec330, 0xc82000a460, 0x2, 0x2, 0x0, 0x0, 0x0, 0x28)
	/home/sean/go/src/github.com/lxc/lxd/lxc/main.go:233 +0xb2d
main.execIfAliases(0xc8200ec330, 0xc82000a460, 0x2, 0x2)
	/home/sean/go/src/github.com/lxc/lxd/lxc/main.go:245 +0x65
main.run(0x0, 0x0)
	/home/sean/go/src/github.com/lxc/lxd/lxc/main.go:98 +0xac0
main.main()
	/home/sean/go/src/github.com/lxc/lxd/lxc/main.go:21 +0x27

```

--- a/lxc/main.go
+++ b/lxc/main.go
@@ -190,26 +190,19 @@ func expandAlias(config *lxd.Config, origArgs []string) ([]string, bool) {
 	aliasValue := []string{}
 
 	for k, v := range config.Aliases {
-		matches := false
-		for i, key := range strings.Split(k, " ") {
-			if len(origArgs) <= i+1 {
-				break
-			}
-
-			if origArgs[i+1] == key {
-				matches = true
-				aliasKey = strings.Split(k, " ")
-				aliasValue = strings.Split(v, " ")
-				break
-			}
-		}
-
-		if !matches {
-			continue
-		}
-
 		foundAlias = true
-		break
+		for i, key := range strings.Split(k, " ") {
+			if len(origArgs) <= i+1 || origArgs[i+1] != key {
+				foundAlias = false
+				break
+			}
+		}
+
+		if foundAlias {
+			aliasKey = strings.Split(k, " ")
+			aliasValue = strings.Split(v, " ")
+			break
+		}
 	}
 
 	if !foundAlias {

